web/src/components/Accordion.tsx

```diff
@@ -34,18 +39,45 @@ export default function Accordion({
   setState: (isCollapsed: boolean) => void;
   isTopExpanding?: boolean;
 }) {
+  const { ref, height: observerHeight } = useResizeObserver<HTMLDivElement>();
+  const [renderChildren, setRenderChildren] = useState(!isCollapsed);
+
+  const [spring, api] = useSpring(
+    () => ({
+      height: isCollapsed ? 0 : observerHeight,
+      // eslint-disable-next-line unicorn/no-nested-ternary -- it interferes with prettier
+      rotate: isCollapsed ? (isTopExpanding ? -90 : 90) : isTopExpanding ? 90 : -90,
+    }),
+    [isCollapsed, isTopExpanding, observerHeight]
+  );
+
   const handleToggleCollapse = () => {
     onClick?.();
 
     isCollapsed && onOpen?.();
 
+    setRenderChildren(true);
+
+    api.start({
+      to: {
+        height: isCollapsed ? observerHeight : 0,
+        // eslint-disable-next-line unicorn/no-nested-ternary -- it interferes with prettier
+        rotate: isCollapsed ? (isTopExpanding ? 90 : -90) : isTopExpanding ? -90 : 90,
+      },
+      onRest: () => {
+        setRenderChildren(isCollapsed);
+      },
+    });
     setState(!isCollapsed);
   };
 
-  const Icon: LucideIcon = isTopExpanding === isCollapsed ? collapsedIcon : expandedIcon;
+  const Icon: LucideIcon =
+    (collapsedIcon && expandedIcon && isTopExpanding === isCollapsed
```

Comment:
This expression is challenging to parse. Consider adding some form of documentation, i.e. comment, saving subexpressions into a well named variable, extracted into a function, etc

Reply:
Really just a work around for the custom icons, provided to the component. I'll make sure to document it though and I've also talked with Alex about as well 🙂